### DISCUSSION TITLE
Possibility to created account with invalide username

### DISCUSSION THREAD

**Discussion by bryan-brancotte** [kjolley/BIGSdb#771] [General]
Dear Keith

Multiple accounts in our instance have username not valid : with space, accent, hyphen... I thought it was a former possibility of BIGSdb, but every year since 2014 accounts like this are created, not in 2021 yet though. I think these accounts could have been created when data were submitted on behalf of someone : I also saw accounts than have a curator field indicating an account that is not a curator. I stumble on this invalide username when working with [move_user.pl](https://github.com/kjolley/BIGSdb/blob/develop/scripts/maintenance/move_user.pl)

Best wishes

**Comment by kjolley** [comment]
If you manually create accounts then the username can be any string up to 12 charcters in length. There is no reason, as far as the software is concerned, that usernames should not have spaces, accents or hyphens. The user status is set by whoever created the account (who must have at least curator access to do so).

For users that sign themselves up using site-wide accounts though, then we restrict them to only alphanumeric characters (A-Z, a-z, 0-9) with no spaces, hyphens or punctuation. I don't think the move_user.pl script does any checks and you should be able to pass the username if you enclose it in single quotes.

**Comment by bryan-brancotte** [comment]
As for the self-registration it is restricted to only alphanumeric characters I though it would be the case for the rest of the app and maybe a test wasn't working.

For the space, indeed enclosing username between quotes helps, but it doesn't with accent, the script keep saying that user is not in the DB. Probably an encoding issue between the system and the database but as we chose to bulk normalize username with [postgres's unaccent](https://www.postgresql.org/docs/9.1/unaccent.html) so it is not a problem anymore.
